tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -663,3 +814,130 @@ func disableQinQOnSwitch(switchCredentials *sriovenv.SwitchCredentials, switchIn
+}
+
+func defineAndCreateServerDPDKPod(
+	podName,
+	nodeName string,
+	serverPodNetConfig []*multus.NetworkSelectionElement,
+	podCmd []string) *pod.Builder {
+	var rootUser int64 = 0
+	securityContext := corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"},
+		},
+	}
+
+	dpdkContainer := pod.NewContainerBuilder(podName, NetConfig.DpdkTestContainer, podCmd)
+	dpdkContainerCfg, err := dpdkContainer.WithSecurityContext(&securityContext).
+		WithResourceLimit("2Gi", "1Gi", 4).
+		WithResourceRequest("2Gi", "1Gi", 4).WithEnvVar("RUN_TYPE", "testcmd").
+		GetContainerCfg()
+
+	Expect(err).ToNot(HaveOccurred(), "Fail to set dpdk container")
+
+	dpdkPod := pod.NewBuilder(APIClient, podName, tsparams.TestNamespaceName,
+		NetConfig.DpdkTestContainer).WithSecondaryNetwork(serverPodNetConfig).
+		DefineOnNode(nodeName).RedefineDefaultContainer(*dpdkContainerCfg).WithHugePages()
+	configMapData := map[string]string{"cmd_file": dpdkPort0Cmd}
+	configMap, err := configmap.NewBuilder(APIClient, "dpdk-port-cmd", tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+	dpdkPod.WithLocalVolume(configMap.Definition.Name, "/etc/cmd")
+	dpdkPod, err = dpdkPod.CreateAndWaitUntilRunning(10 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Fail to create server pod")
+
+	return dpdkPod
+}
+
+func defineAndCreateClientDPDKPod(
+	podName,
+	nodeName string,
+	serverPodNetConfig []*multus.NetworkSelectionElement) *pod.Builder {
+	var rootUser int64 = 0
+	securityContext := corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"},
+		},
+	}
+
+	dpdkContainer := pod.NewContainerBuilder(podName, NetConfig.DpdkTestContainer,
+		[]string{"/bin/bash", "-c", "sleep INF"})
+	dpdkContainerCfg, err := dpdkContainer.WithSecurityContext(&securityContext).
+		WithResourceLimit("2Gi", "1Gi", 4).
+		WithResourceRequest("2Gi", "1Gi", 4).WithEnvVar("RUN_TYPE", "testcmd").
+		GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Fail to set dpdk container")
+
+	dpdkPod := pod.NewBuilder(APIClient, podName, tsparams.TestNamespaceName,
+		NetConfig.DpdkTestContainer).WithSecondaryNetwork(serverPodNetConfig).
+		DefineOnNode(nodeName).RedefineDefaultContainer(*dpdkContainerCfg).WithHugePages()
+
+	dpdkPod, err = dpdkPod.CreateAndWaitUntilRunning(10 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Fail to create server pod")
+
+	return dpdkPod
+}
+
+func rxTrafficOnClientPod(clientPod *pod.Builder, clientRxCmd string) {
+	Expect(clientPod.WaitUntilRunning(time.Minute)).ToNot(HaveOccurred(),
+		"Fail to wait until pod is running")
+
+	clientOut, err := clientPod.ExecCommand([]string{"/bin/bash", "-c", clientRxCmd})
+
+	var timeoutError = "command terminated with exit code 137"
+	if err.Error() != timeoutError {
```

Comment:
if err.Error() != "command terminated with exit code 137"

Reply:
checking

Reply:
?

Reply:
The following code is from the rootless test case. It has been running consistently on those test cases.
If you prefer we can move the Expect out of the if condition and change the if condition to equal a non empty err with description.  

if err.Error() != timeoutError {
		Expect(err).ToNot(HaveOccurred(), "Fail to exec cmd")
	}